### PR TITLE
update readme with docker instructions and other miscellaneous updates

### PR DESCRIPTION
**Pull request by ExtraGravity**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -5,29 +5,63 @@
 Join us on Discord!
 https://discord.gg/46vtdY
```
Comment (ctlai95) [inline README.adoc]: We should remove/update this because the link expired

File: README.adoc [hunk 2]
```diff
@@ -5,29 +5,63 @@
...
+To build TerminalFighter and run its tests, run `build.sh` with the `src` command.
+
+	./build.sh src
+
+Aside from runtime dependencies, the following are required to build Terminal Fighter:
+
+* CMake
+* A C++17 compiler
+
+Other commands available are:
```
Comment (ctlai95) [inline README.adoc:24]: What about the `all` command?

File: README.adoc [hunk 3]
```diff
@@ -5,29 +5,63 @@
...
+
+* `format` will lint all the source code and format it according to the `astylerc` file, `astyle` will need to be installed for this.
+
+* `docs` build documentation out of the inline comments in the source using Doxygen.
+* `clean` will remove any old build artifacts.
+
+=== Dependencies
+
+A number of dependencies are required to build Terminal Fighter, a non exhaustive list includes.
```
Comment (ctlai95) [inline README.adoc]: `includes:` would be better

**Review by ctlai95 — CHANGES_REQUESTED** [review]

File: README.adoc (hunk 2 above)
Comment (ExtraGravity) [reply README.adoc:24]: Added

**Review by ctlai95 — APPROVED** [review]

**Pull request merged**